src/net.h

```diff
@@ -812,19 +818,21 @@ class CNode
 
     void AddAddressKnown(const CAddress& _addr)
     {
-        addrKnown.insert(_addr.GetKey());
+        LOCK(m_addr_relay.cs_addrsend);
+        m_addr_relay.addrKnown.insert(_addr.GetKey());
     }
 
     void PushAddress(const CAddress& _addr, FastRandomContext &insecure_rand)
     {
+        LOCK(m_addr_relay.cs_addrsend);
         // Known checking here is only to save space from duplicates.
         // SendMessages will filter it again for knowns that were added
         // after addresses were pushed.
-        if (_addr.IsValid() && !addrKnown.contains(_addr.GetKey())) {
-            if (vAddrToSend.size() >= MAX_ADDR_TO_SEND) {
-                vAddrToSend[insecure_rand.randrange(vAddrToSend.size())] = _addr;
+        if (_addr.IsValid() && !m_addr_relay.addrKnown.contains(_addr.GetKey())) {
```

Comment:
https://github.com/bitcoin/bitcoin/pull/15759/commits/c36939ca2ea636dce8047065d88c2e70bd418405

nit, coult early return before the above lock:
```cpp
if (!_addr.IsValid()) return;
LOCK(m_addr_relay.cs_addrsend);
if (!m_addr_relay.addrKnown.contains(_addr.GetKey())) {
```

Reply:
Done.